### PR TITLE
Add Supervisor logs to core page

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change
CleanShot 2022-04-25 at 10 04 35

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/logs/error-log-card.ts [hunk 1]
```diff
@@ -1,102 +1,185 @@
...
+import "../../../components/ha-select";
 import { fetchErrorLog } from "../../../data/error_log";
+import { extractApiErrorMessage } from "../../../data/hassio/common";
+import { fetchHassioLogs } from "../../../data/hassio/supervisor";
 import { HomeAssistant } from "../../../types";
 
+interface LogProvider {
+  key: string;
+  name: string;
+}
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:26]: Move to `data`

File: src/panels/config/logs/error-log-card.ts [hunk 2]
```diff
@@ -1,102 +1,185 @@
...
     `;
   }
 
-  protected firstUpdated(changedProps) {
+  protected firstUpdated(changedProps: PropertyValues) {
     super.firstUpdated(changedProps);
 
+    if (isComponentLoaded(this.hass, "hassio")) {
+      this._hassioLoaded = true;
+    }
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:126]: Not sure if `hassio` is always already loaded when this page is first updated, would probably keep this in `render` and not in firstupdated

File: src/panels/config/logs/error-log-card.ts [hunk 3]
```diff
@@ -1,102 +1,185 @@
...
   @property() public filter = "";
 
   @state() private _isLogLoaded = false;
 
-  @state() private _errorHTML!: TemplateResult[] | string;
+  @state() private _logHTML!: TemplateResult[] | string;
+
+  @state() private _error?: string;
+
+  @state() private _selectedLogProvider = "core";
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:67]:
```suggestion
  @state() private _selectedLogProvider?: string;
```

File: src/panels/config/logs/error-log-card.ts [hunk 4]
```diff
@@ -1,102 +1,185 @@
...
+
+  @state() private _hassioLoaded = false;
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:69]:
```suggestion
```

File: src/panels/config/logs/error-log-card.ts [hunk 5]
```diff
@@ -1,102 +1,185 @@
...
-      .warning {
-        color: var(--warning-color);
-      }
+  private async _refreshLogs(): Promise<void> {
+    this._logHTML = this.hass.localize("ui.panel.config.logs.loading_log");
+    let log: string;
 
-      :host-context([style*="direction: rtl;"]) mwc-button {
-        direction: rtl;
+    if (this._hassioLoaded) {
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:163]:
```suggestion
    if (this._selectedLogProvider) {
```

File: src/panels/config/logs/error-log-card.ts [hunk 6]
```diff
@@ -1,102 +1,185 @@
...
+import { extractApiErrorMessage } from "../../../data/hassio/common";
+import { fetchHassioLogs } from "../../../data/hassio/supervisor";
 import { HomeAssistant } from "../../../types";
 
+interface LogProvider {
+  key: string;
+  name: string;
+}
+
+const logProviders: LogProvider[] = [
```
[the remark this answers is not in the document]
Comment (zsarnett) [reply src/panels/config/logs/error-log-card.ts:23]: probably. Not sure where to do that from? Supervisor doesn't have an API for that it seems

[the remark this answers is not in the document]
Comment (bramkragten) [reply src/panels/config/logs/error-log-card.ts:23]: No, that should be added then

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/logs/error-log-card.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (ludeeus) [reply src/panels/config/logs/error-log-card.ts:23]: https://github.com/home-assistant/supervisor/issues/3295

**Pull request merged**